consensus/service/src/config/tokens.rs

```diff
@@ -39,10 +104,73 @@ impl TokenConfig {
         self.minimum_fee
             .or_else(|| FeeMap::default().get_fee_for_token(&self.token_id()))
     }
+
+    /// Master minters config, when available.
+    pub fn master_minters(&self) -> Option<&SignerSet<Ed25519Public>> {
+        // Can never have master minters for MOB
+        if self.token_id == TokenId::MOB {
+            return None;
+        }
+        self.master_minters.as_ref()
+    }
+
+    /// Check if the token configuration is valid.
+    pub fn validate(&self) -> Result<(), ConsensusServiceError> {
+        // We must have a fee for every configured token.
+        if self.minimum_fee_or_default().is_none() {
+            return Err(ConsensusServiceError::Configuration(
+                "missing minimum fee".to_string(),
+            ));
+        }
+
+        // By default, we restrict MOB minimum fee to a sane value.
+        if self.token_id == Mob::ID {
+            let mob_fee = self.minimum_fee_or_default().unwrap(); // We are guaranteed to have a minimum fee for MOB.
+            if !self.allow_any_fee && !(10_000..1_000_000_000_000u64).contains(&mob_fee) {
+                return Err(ConsensusServiceError::Configuration(format!(
+                    "Fee {} picoMOB is out of bounds",
+                    mob_fee
+                )));
+            }
+        } else {
+            // allow_any_fee can only be used for MOB
+            if self.allow_any_fee {
```

Comment:
Optional/Equivalent: else if

Reply:
I find it easier to read with the nesting, and it makes it a tiny bit easier to add more non-MOB validation (like if we making minting a requirement for example)